packages/next/src/server/next.ts

```diff
@@ -287,9 +355,11 @@ class NextCustomServer extends NextServer {
       minimalMode: this.options.minimalMode,
       quiet: this.options.quiet,
     })
-    this.requestHandler = initResult[0]
-    this.upgradeHandler = initResult[1]
-    this.renderServer = initResult[2]
+    this.init = {
```

Comment:
The init result is already in an array here, maybe the type is just the return type of the `getRequestHandlers`?

Reply:
you're saying we keep it as a tuple and do `this.init[0]`, `this.init[1]`, `this.init[1]`? honestly not a fan, tbh i really dislike this tuple thing and wanted to get rid of it, so i'd rather change `getRequestHandlers` to return an object 😅 wdyt?

Reply:
Let's make it not return a tuple